sherpa_contrib/diag_resp.py

```diff
@@ -0,0 +1,908 @@
+#
+#  Copyright (C) 2024
+#           Smithsonian Astrophysical Observatory
+#
+#
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation; either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program; if not, write to the Free Software
+# Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA
+# 02110-1301 USA.
+#
+
+
+
+
+""" Create diagonal RMF and flat ARF objects of value unity.
+
+Helper routines for creating a diagonal RMF and flat ARF response
+objects for various telescopes and instruments.
+The default is to generate Chandra/ACIS responses for PI-type spectra
+extracted from ACIS stowed background event files.
+
+Examples
+--------
+
+>>> from sherpa_contrib.diag_resp import *
+>>> diagrmf,flatarf = mkdiagresp()
+>>>
+>>> set_rmf(diagrmf, bkg_id=1)
+>>> set_arf(flatarf, bkg_id=1)
+
+or
+
+>>> diagrmf,flatarf = mkdiagresp(refspec="bkg.pi")
+
+or
+
+>>> bkgspec = get_bkg()
+>>> diagrmf,flatarf = mkdiagresp(refspec=bkgspec)
+
+"""
+
+__revision__ = "22 July 2024"
+
+import os
+import warnings
+from logging import getLogger
+from functools import wraps
+from re import sub
+
+import numpy.typing as npt
+from numpy import arange
+
+from ciao_contrib._tools.fileio import get_keys_from_file
+
+from sherpa.astro.data import DataPHA
+from sherpa.astro.ui import create_rmf as shpmkrmf
+from sherpa.astro.ui import create_arf as shpmkarf
+
+
+
+def _reformat_wmsg(func):
+    """
+    Decorator:
+
+    Trim extraneous user-warning information showing path and line number
+    where Sherpa warning message is being thrown for replacing 0s with ethresh.
+    """
+
+    @wraps(func)
+    def run_func(*args, **kwargs):
+        warnings_reset = warnings.formatwarning
+        warnings.formatwarning = lambda msg, *args_warn, **kwargs_warn: f'Warning: {msg}\n'
+
+        func_out = func(*args,**kwargs)
+
+        warnings.formatwarning = warnings_reset
+
+        return func_out
+
+    return run_func
+
+
+
+def _quash_shpverb(func):
```

Comment:
You should be able to do this with just

```
from sherpa.utils.logging import SherpaVerbosity
...
with SherpaVerbosity("ERROR"):
    ... call code ...
```

So, either this can be used in the decorator, or you can avoid the decorator and use the form above (I haven't checked how it's used).

This is not something that needs to be done now, since the code works as is.